cmake/kokkos_arch.cmake

```
@@ -433,8 +433,14 @@ CHECK_AMDGPU_ARCH(VEGA906 gfx906) # Radeon Instinct MI50 and MI60
 CHECK_AMDGPU_ARCH(VEGA908 gfx908)
 
 IF(KOKKOS_ENABLE_HIP AND NOT AMDGPU_ARCH_ALREADY_SPECIFIED)
-  MESSAGE(SEND_ERROR "HIP enabled but no AMD GPU architecture currently enabled. "
-                     "Please enable one AMD GPU architecture via -DKokkos_ARCH_{..}=ON'.")
+  FIND_PATH(ROCM_ENUMERATOR_PATH rocm_agent_enumerator)
+  EXECUTE_PROCESS(COMMAND "${ROCM_ENUMERATOR_PATH}/rocm_agent_enumerator" OUTPUT_VARIABLE GPU_ARCHS)
+  STRING(LENGTH "${GPU_ARCHS}" len_str)
+  # enumerator always output gfx900 as the first line
```

```suggestion
  # enumerator always outputs gfx000 as the first line
```
? That's at least what I am seeing.